jvm/core/src/main/kotlin/com/intuit/player/jvm/core/player/state/PlayerFlowState.kt

```diff
@@ -99,8 +99,13 @@ internal class ErroneousState(override val node: Node) :
 
     override val flow: Flow get() = node.getSerializable("flow", Flow.serializer()) ?: Flow()
 
-    override val error: PlayerException get() = node.getSerializable("error", PlayerException.serializer())
-        ?: PlayerException(node.getString("error") ?: "unable to determine error")
+    override val error: PlayerException get() = when (val rawError = node["error"]) {
+        is PlayerException -> rawError
+        is Exception -> PlayerException(rawError.message ?: "", rawError)
```

Comment:
out of curiosity, when do these two cases come into play

Reply:
It's more for completeness than anything. I actually think that Graal works this way and is able to spit up the original exception, which may or may not be a `PlayerException`, but that could be wrong. If `Node` internals change in the future with some introspection for `Node` children that are error-like and try to deserialize automatically, this will account for that.